modules/EmailMan/EmailMan.php

```diff
@@ -628,7 +628,7 @@ public function sendEmail($mail, $save_emails = 1, $testmode = false)
         if (
                 $module instanceof Basic &&
                 $sugar_config['email_enable_confirm_opt_in'] &&
-                !$module->getEmailAddressConfirmOptIn('email1')
+                empty($module->getEmailAddressConfirmOptIn('email1')->confirm_opt_in)
```

Comment:
Non-logical using of `getEmailAddressConfirmOptIn()` method. It return an Email Address instead Confirm Opt-In status which has a property named confirm_opt_in?

Reply:
I see what you mean. What if change this to getEmailAddressFromSugarField instead?

Reply:
I meant, if $module instance of Basic and the Basic class implemented a getEmailAddressConfirmOptIn() properly as it was then it should not changed.